### DISCUSSION TITLE
Support for colorable shields

### DISCUSSION THREAD

**Discussion by SenorRagequit** [exorstudios/riftbreaker-tools#134] [Wishlist] [answered]
Asked by WirawanMYT [Migrated from issues]

**Wish**
---
Support for colorable shields defined in .material similar to how laser beam .mesh are colored.

**Why**
---
For easier visual differentiation between different types of shield entities. At the moment, the shield colour properties are hardcoded in barrier.hpsl (?) as shown by Lukaasm in rb-modding channel. The workaround involves defining the shield cGlowEmissive (for now) in LUA, but this is inflexible and not conducive for rapid testing. It would be also nice that different shield entities retain their respective shield colour and glow instead of locked to one colour properties for all different shield entities.

**Comment by lukaasm** [comment, marked as answer]
```
ShieldComponent
{
    highlight_material "highlight/shield"
}
```

You will be able to specify custom highlight material in ShieldComponent int the future
